scipy/stats/_qmc.py

```diff
@@ -1004,29 +1005,42 @@ class LatinHypercube(QMCEngine):
     >>> qmc.discrepancy(sample)
     0.0176...  # random
 
-    Use the `orthogonal` keyword argument to produce an orthogonal array based
-    LHS. The construction imposes to use a square of a prime number.
+    Use the `strength` keyword argument to produce an orthogonal array based
+    LHS of strength 2. The construction imposes to use a square of a prime
+    number.
 
-    >>> sampler = qmc.LatinHypercube(d=2, orthogonal=True)
+    >>> sampler = qmc.LatinHypercube(d=2, strength=2)
     >>> sample = sampler.random(n=9)
     >>> qmc.discrepancy(sample)
     0.00526...  # random
 
     Options could be combined as a same time to produce an optimized centered
     orthogonal array based LHS. After optimization, the result would not
-    be guaranteed to be orthogonal.
+    be guaranteed to be of strength 2.
 
     """
 
     def __init__(
         self, d: IntNumber, *, centered: bool = False,
-        orthogonal: bool = False,
+        strength: int = 1,
         optimization: Optional[Literal["random-cd"]] = None,
         seed: SeedType = None
     ) -> None:
         super().__init__(d=d, seed=seed)
         self.centered = centered
 
+        lhs_method_strength = {
+            1: self._random,
+            2: self._random_oa_lhs
+        }
+
+        try:
+            self.lhs_method = lhs_method_strength[strength]
+        except KeyError as exc:
+            raise ValueError(f"{strength!r} is not a valid"
+                             " strength. It must be one of"
+                             f" {set(lhs_method_strength)!r}") from exc
```

Comment:
This is one of those places where defining `message` first might look cleaner and reduce lines. (but not required for merge)

Reply:
Done and elsewhere.